### ISSUE TITLE
 mixed-precision training for QuartzNet

### ISSUE THREAD

**Issue by ArtemisZGL** [NVIDIA/NeMo#599]
I used the lirbrispeech clean 100 dataset to train the quartznet. And it's normal without mixed precision, getting the wer for test-clean is 16.36%, but when I used the O1 flag for mixed precision keeping other param the same, it showed " Gradient overflow. Skipping step, loss scaler 0 reducing loss scale to 1.0" after about 10 epoch(this seems happen in different time everytime I run). Before that, everthing seems normal, the training wer and evaluation wer is decrease normally. 
![image](https://user-images.githubusercontent.com/31615877/80680212-e8c17c80-8af0-11ea-8e32-e9ec300863ac.png)  

Did I miss something for mixed precision training ? why the loss will overflow suddenly?

**Comment by okuchaiev** [comment]
Can you try increasing warmup_steps? Something like 16,000 or 8,000.
Also, when you train with O1 you should be able to fit about 2x larger batch size compared to O0. So use the maximum batch size you can fit per GPU when using O1 - this should both help numerical stability and make training faster.

**Comment by ArtemisZGL** [comment]
@okuchaiev i will try that, thanks. Beside,  I train the LS + LS speed perturbed data with the param you said in https://github.com/NVIDIA/NeMo/issues/230 with flag O0, and the batch size i can set is 16, so the iter per step param i set to 2, but the training didn't converge after 3days,  wer is always larger than 100%, did I miss something ？

**Comment by ArtemisZGL** [comment]
@okuchaiev once I add the warmup_steps to 8000 or 16000, the loss can converge now, both the mixed-precision training and the  LS  training I mentioned above. Thanks you so much ! Besides, can I  ask one more question about the param used for finetune, which is not mentioned in the paper when you finetune the WSJ dataset.

**Comment by okuchaiev** [comment]
>>"can I ask one more question about the param used for finetune" 
- sure. What questions do you have? For fine-tuning, we typically use learning rate 10x smaller compared to the one when training from scratch.  
Also, we are working on a fine-tuning paper, should appear on arxiv next week,  I'll post a link here

**Comment by ArtemisZGL** [comment]
@okuchaiev Thanks, I will try the 10x smaller lr to finetune. Looking forward to your paper!

**Comment by DennisMuellerKenbun** [comment]
Hey guys,

I ran into the same problem here. So, thanks for the tip. Do you have any intuition why and when more warmup steps are helping?

Thx.

**Comment by ArtemisZGL** [comment]
I have read the paper https://arxiv.org/abs/2005.04290,. I have a question, I download the african accent dataset CORAAL, but I found if I only use the african accent part, the hour is less than 88 mentioned in the paper, have you use the interviewer part as well ? @okuchaiev

**Comment by redoctopus** [comment]
Yes, all speech data in the CORAAL dataset was included.

**Comment by ArtemisZGL** [comment]
@redoctopus thanks, have you filter those segment with short duration?

**Comment by redoctopus** [comment]
We first combined consecutive segments with the same speaker to reach a target minimum duration of 3s. Then we filtered out any segments longer than 16.7s and shorter than 0.1s.

**Comment by ArtemisZGL** [comment]
@redoctopus Thanks !
